sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -140,3 +142,41 @@ AZ_NODISCARD az_result az_iot_hub_client_properties_append(
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result az_iot_hub_client_properties_find(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span* out_value)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(name, 1, false);
+  AZ_PRECONDITION_NOT_NULL(out_value);
+
+  az_span remaining = properties->_internal.properties;
+
+  while (az_span_length(remaining) != 0)
+  {
+    az_span delim_span = az_span_token(remaining, hub_client_param_equals_span, &remaining);
+    if (az_span_is_content_equal(delim_span, name))
+    {
+      *out_value = az_span_token(remaining, hub_client_param_separator_span, &remaining);
+      return AZ_OK;
+    }
+    else
+    {
+      az_span value;
+      value = az_span_token(remaining, hub_client_param_separator_span, &remaining);
+      (void)value;
+      if (az_span_length(remaining) > 0)
```

Comment:
Is this check duplicated from line 157?
`while (az_span_length(remaining) != 0)`

Reply:
Not exactly. Note on line 168 how remaining is used as the output span. When it is passed to token, it becomes everything after token. Example of `az_span_token`:
```
Token: &
Search string: key=value&my_thing=my_value
Return: [key=value]&<my_thing=my_value>
Where [] is the returned span and <> is the output span. 
On line 168 returned span is "value" and output span is "&remaining"
```

Reply:
I think Eric might have had the same comment as you below. I removed this if/else logic to simplify. Thanks for the tip Tiejun